lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -573,10 +609,20 @@ export class UserAgentApplication {
      *
      */
     acquireTokenSilent(request: AuthenticationParameters): Promise<AuthResponse> {
-        if (!request) {
-            throw ClientConfigurationError.createEmptyRequestError();
-        }
+
+        // generate state
+        const state = UrlUtils.getRequestState(request && request.state);
+
         return new Promise<AuthResponse>((resolve, reject) => {
+            // throw on null request
+            if (!request) {
+                throw ClientConfigurationError.createEmptyRequestError();
+            }
+
+            // return an error if called from the hidden iframe created by the msal js silent calls
+            if (UrlUtils.urlContainsHash(window.location.hash) && WindowUtils.isInIframe()) {
+                throw ClientConfigurationError.createBlockTokenRequestsInHiddenIframeError();
```

Comment:
This should be `reject(ClientConfigurationError.createBlockTokenRequestsInHiddenIframeError())` so that it returns a rejected promise.